### PR TITLE
Added LHUC for transformer model

### PR DESCRIPTION
**Pull request by davvil**
Note: Changed apply() function of LHUC to __call__

#### Pull Request Checklist ##
- [x] Changes are complete
- [x] Unit tests pass (`pytest`)
- [x] System tests pass (`pytest test/system`)
- [x] Passed code style checking (`./style-check.sh`)
- [x] You have considered writing a test
- [x] Updated major/minor version in `sockeye/__init__.py`. Major version bump if this is a backwards incompatible change.
- [x] Updated CHANGELOG.md

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: sockeye/arguments.py [hunk 1]
```diff
@@ -646,6 +636,18 @@ def add_model_parameters(params):
...
+    # LHUC
+    # TODO: The convolutional model does not support lhuc yet
+    model_params.add_argument('--lhuc',
+                              nargs="+",
+                              default=None,
+                              choices=C.LHUC_CHOICES,
+                              metavar="COMPONENT",
+                              help="Use LHUC (Vilar 2018). Include an amplitude parameter to hidden units for"
+                              " domain adaptation. Needs a pre-trained model. Valid values: {values}. Currently only"
+                              " supported for RNN models. Default: %(default)s.".format(
```
Comment (fhieber) [inline sockeye/arguments.py]: help string is not accurate

File: sockeye/layers.py [hunk 2]
```diff
@@ -113,7 +113,7 @@ def __init__(self,
         else:
             self.params = weight
 
-    def apply(self,
+    def __call__(self,
               inputs: mx.sym.Symbol,
```
Comment (fhieber) [inline sockeye/layers.py]: indendation

File: sockeye/train.py [hunk 3]
```diff
@@ -101,8 +101,11 @@ def check_arg_compatibility(args: argparse.Namespace):
                         % (args.transformer_model_size, args.num_embed[1]))
 
     if args.lhuc is not None:
-        check_condition(args.encoder == C.RNN_NAME or args.decoder == C.RNN_NAME,
+        # Actually this check is a bit too strict
+        check_condition(args.encoder != C.CONVOLUTION_TYPE or args.decoder != C.CONVOLUTION_TYPE,
                         "LHUC is only supported for RNN models for now.")
```
Comment (fhieber) [inline sockeye/train.py]: incorrect error message

**Comment by davvil** [comment]
Ups! I had this on my mind and then it slipped off. Updating now.

**Pull request merged**